Split session statics reset method to prevent unloading seasonal backgrounds

dekrain opened the pull request:
Fixes #16180

dekrain committed:
Split session statics reset method to prevent unloading seasonal backgrounds while idle

bdach commented on osu.Game/Configuration/SessionStatics.cs:
```diff
@@ -24,6 +22,18 @@ public void ResetValues()
+            ensureDefault(GetBindable<bool>(Static.LoginOverlayDisplayed));
+            ensureDefault(GetBindable<bool>(Static.MutedAudioNotificationShownOnce));
+            ensureDefault(GetBindable<bool>(Static.LowBatteryNotificationShownOnce));
+            ensureDefault(GetBindable<double?>(Static.LastHoverSoundPlaybackTime));
```
Arguably `LastHoverSoundPlaybackTime` also shouldn't be reset. The inactivity period required is long enough to not matter here, but from a semantic point of view there's no reason to do anything with hover sound debounce logic if the user has been deemed inactive.

Interested in more thoughts on this in general from other reviewers.

peppy replied:
Can agree to this. Would probably also rename this method to be something like `ResetAfterInactivity`.

bdach replied:
Removed the resetting of `LastHoverSoundPlaybackTime` myself in latest batch of commits.

peppy committed:
Rename method and simplify `SetDefault` calling code

peppy committed:
Fix `SessionStaticsTest` never working

In headless operation, the test would complete before ever reaching an
idle state. And this test only ran headless..

peppy commented:
I've made some refactors to the naming and the structure of the code. Also fixed the related test not actually... working.. ever.

bdach committed:
Remove unused private method

bdach committed:
Do not reset last hover sound playback time on inactivity

bdach committed:
Cover seasonal backgrounds not resetting on inactivity in test

bdach commented:
I've piled on some of my own changes onto this branch, like removing an unused method, not resetting the `LastHoverSoundPlaybackTime` static and adding test coverage for the fail case here. Looks good otherwise.
